runtime/str.go

```diff
@@ -157,6 +157,32 @@ func strAdd(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return NewStr(stringV + stringW).ToObject(), nil
 }
 
+func strCapitalize(f *Frame, args Args, kwargs KWArgs) (*Object, *BaseException) {
+	if raised := checkMethodArgs(f, "capitalize", args, StrType); raised != nil {
+		return nil, raised
+	}
+	s := toStrUnsafe(args[0])
+	sv := s.Value()
+	numBytes := len(sv)
+	if numBytes == 0 {
+		return s.ToObject(), nil
+	}
+	buf := bytes.Buffer{}
```

Comment:
`bytes.Buffer{}` might be overkill here. You could preallocate a []byte of the correct size and write to that directly.